Juliet C# upgrade to net6.0 version!

DanielELog opened the pull request:
I have yet to skim through the projects myself and verify the logic is kept intact. Nevertheless, it is up for grabs and discussion!

DanielELog committed:
[feat] adds script updating Juliet projects

DanielELog committed:
[feat] updates Juliet to net6.0

misonijnik commented:
Wow, what a fantastic piece of work! It looks awesome to me. Thank you!

To make it even better, could you add some simple CI to check the build on .NET 6.0?

I will review your PR after I get the build checked.

Thanks again!

DanielELog committed:
[feat] adds basic ci

DanielELog committed:
[fix] improves build workflow

- case-sensitive directory name change
- error throw upon the first build fail occurrence

DanielELog commented:
Hello there!

Added basic CI and tested it on my forked repo - you can inspect the run here: https://github.com/DanielELog/JulietCSharp-mutated/actions/runs/9650216146/job/26615384183?pr=1

DanielELog committed:
[chore] adds README

DanielELog committed:
[chore] adds LICENSE

misonijnik commented on README.md:
```diff
@@ -0,0 +1,3 @@
+# JulietC# mutated
```
JulietC# -> Juliet C#

misonijnik commented on README.md:
```diff
@@ -0,0 +1,3 @@
+# JulietC# mutated
+
+**JulietC# mutated** is aimed to create an extensive C# Benchmark for SAST tool evaluation. It is based on <a href="https://samate.nist.gov/SARD/test-suites/110">Juliet C# 1.3</a> and is currently undergoing active development.
```
JulietC# -> Juliet C#

misonijnik replied:
"is aimed" -> "aims"

misonijnik commented on src/TestCaseSupport/TestCaseSupport/.updated:
Is it necessary to have `.updated` files?

DanielELog replied:
The upgrading script is not able to differentiate between old and new versions of each single project without them. They can be left out, but then if the script is ever called again, it will be confused.

misonijnik commented on .gitignore:
Ignore only those paths related to the repository.

DanielELog replied:
Could you please explain what paths you meant?

DanielELog committed:
[chore] tweaks readme style

DanielELog committed:
[fix] removes unnecessary files
